src/invocation/ClientConnection.ts

```diff
@@ -66,6 +66,13 @@ class ClientConnection {
             this.socket = netSocket as stream.Duplex;
         }
 
+        netSocket.setTimeout(this.clientNetworkConfig.connectionTimeout)
+        this.socket.on('timeout', (e: any) => {
```

Comment:
As far as I see from Node.js documentation, timeout event callback is not called with an argument. Therefore `e` is always undefined.
It would be better if we could pass the error you logged to the promise and `HazelcastClient.newHazelcastClient()` would fail with this error.